File: 2848-points-that-intersect-with-cars/2848-points-that-intersect-with-cars.cpp
class Solution {
public:
    int numberOfPoints(vector<vector<int>>& nums) {
    int line [102] ={};
    for(auto& p : nums){
        line[p[0]]++;
        line[p[1]+1]--;
    }
    int ans =0;
    int count =0;
    for(int i =0; i <102; ++i){
        count += line[i];
        if(count > 0){
            ++ans;
        }
    }
    return ans;
}
};

File: 2848-points-that-intersect-with-cars/2848-points-that-intersect-with-cars.py
from collections import defaultdict
from typing import List

class Solution:
    def numberOfPoints(self, nums: List[List[int]]) -> int:
        diff = defaultdict(int)

        for s, e in nums:
            diff[s] += 1
            diff[e + 1] -= 1   # inclusive interval

        covered = 0
        active = 0
        prev = None

        for point, val in sorted(diff.items()):
            # count span since previous point, if covered
            if prev is not None and active > 0:
                covered += point - prev

            active += val
            prev = point

        return covered
